Gems/Atom/RHI/Code/Source/RHI/DrawPacket.cpp

```diff
@@ -47,5 +50,29 @@ namespace AZ
         {
             reinterpret_cast<const DrawPacket*>(p)->m_allocator->DeAllocate(p);
         }
+
+        void DrawPacket::SetRootConstant(uint32_t offset, const AZStd::span<uint8_t>& data)
+        {
+            bool sizeValid = data.size() <= aznumeric_cast<uint32_t>(m_rootConstantSize) - offset;
+            if (sizeValid)
+            {
+                memcpy((void*)(m_rootConstants + offset), data.data(), data.size());
+            }
+            else
+            {
+                AZ_Assert(sizeValid, "New root constants exceed the original size.");
+            }
+        }
+
+        void DrawPacket::SetInstanceCount(uint32_t instanceCount)
+        {
+            for (size_t drawItemIndex = 0; drawItemIndex < m_drawItemCount; ++drawItemIndex)
+            {
+                const DrawItem* drawItemConst = m_drawItems + drawItemIndex;
+                DrawItem* drawItem = const_cast<DrawItem*>(drawItemConst);
```

Comment:
Add a comment as to why we need to drop constness. I.e the need to mutate instance count for mesh instancing.